src/random.h

```diff
@@ -13,12 +13,33 @@
 
 #include <functional>
 #include <limits>
-#include <stdint.h>
+
+#include "int128.h"
 
 /**
  * Functions to gather random data via the rand_core OsRng
  */
 void GetRandBytes(unsigned char* buf, size_t num);
+
+template <typename I>
+I GetRandGeneric(I nMax)
+{
+    static_assert(std::numeric_limits<I>::min() == 0);
+    if (nMax == I())
+        return I();
```

Comment:
This seems like black magic. c++ primitive types have constructors? nonblocking, a comment would be appreciated.

Reply:
Yes they do! Will add.